### PR TITLE
CONCF- 220 Extend full Krux to Mercury

### PR DESCRIPTION
**Pull request by dianafa**
https://wikia-inc.atlassian.net/browse/CONCF-220

After cooperation with Ads team we agreed on moving Krux tracking int Ads module as it will be consistent with new Krux handling proposed in https://github.com/Wikia/app/pull/6605/files. 

Works as complement of https://github.com/Wikia/app/pull/6605. Do not test without that.

### CODE REVIEW

File: front/scripts/mercury/utils/track.ts [hunk 1]
```diff
@@ -153,7 +153,8 @@ module Mercury.Utils {
 	 * trackPageView is called in ArticleView.onArticleChange
 	 */
 	export function trackPageView (adsContext: any) {
-		var trackers: {[name: string]: TrackerInstance} = Em.get('Mercury.Modules.Trackers');
+		var trackers: {[name: string]: TrackerInstance} = Em.get('Mercury.Modules.Trackers'),
+			kruxTracker = new Mercury.Modules.Ads;
```
Comment (rogatty) [inline front/scripts/mercury/utils/track.ts]: It's not really a krux tracker, it's ads module.

File: front/scripts/mercury/utils/track.ts [hunk 2]
```diff
@@ -167,6 +168,9 @@ module Mercury.Utils {
 				trackerInstance.trackPageView(trackerInstance.usesAdsContext ? adsContext : context);
 			}
 		});
+
+	kruxTracker.loadKrux();
```
Comment (rogatty) [inline front/scripts/mercury/utils/track.ts]: I don't like this solution at all. We're introducing an exception here - this whole file is all about trackers and here we're referencing ads module, not a tracker.

If we really want to have Krux integrated with Ads module, can we create some facade Krux tracker that would call Ads module?

File: front/scripts/mercury/modules/Ads.ts [hunk 3]
```diff
@@ -4,7 +4,11 @@
 'use strict';
 
 interface Window {
-	gaTrackAdEvent: any
+	gaTrackAdEvent: any;
+	Krux: {
+		load?: (skinSiteId: string) => void;
+		getParams?: (n: string) => any;
```
Comment (nandy-andy) [inline front/scripts/mercury/modules/Ads.ts]: Oh, we've changed that. 'wikia.krux' module has now three public methods: `load()`, `getUser()` and `getSegments()`.

File: front/scripts/mercury/utils/track.ts (hunk 1 above)
Comment (nandy-andy) [inline front/scripts/mercury/utils/track.ts]: Actually soon [it'll be a Wikia module](https://github.com/Wikia/app/pull/6605) ;) I'd call it just ol'good `krux`.

**Comment by nandy-andy** [comment]
Hi, I just wanted to let you know [our changes](https://github.com/Wikia/app/pull/6605) got tested and merged to Wikia/dev. Ping us if you need help with testing Krux on Mercury :smile_cat:

**Comment by bognix** [comment]
LGTM

**Comment by tomasznapieralski** [comment]
LGTM

**Comment by kenkouot** [comment]
This should not have been merged to dev. It contained numerous TS errors and broken dev and dev-based branches that were branched off once merged. I fixed it in #623

**Pull request merged**